File: day-1/go/solution-a.go
package main

import (
	"fmt"
	"math"
	"os"
	"strconv"
	"strings"
)

func main() {
	contents, err := os.ReadFile("../input.txt")
	if err != nil {
		panic(err)
	}

	lines := strings.Split(string(contents), "\n")

	increments := 0
	lastNumber := math.MaxInt

	for _, line := range lines {
		number, err := strconv.Atoi(line)
		if err != nil {
			panic(err)
		}

		if lastNumber < number {
			increments++
		}
		lastNumber = number
	}

	fmt.Println("Solution:", increments)
}

File: day-1/typescript/src/solution-a.ts
export const solution = "a";

const input = await Bun.file("../input.txt").text();

let increments = 0;
let lastNumber = Infinity;

for (const number of input.split("\n")) {
  if (lastNumber < Number(number)) {
    increments++;
  }
  lastNumber = Number(number);
}

console.log('Solution:', increments);